components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -147,32 +151,29 @@ def create_and_monitor_job_in_db(db_config: Database, wildcard_query: str,
                 job_complete = True
             db_conn.commit()
 
-            time.sleep(1)
+            time.sleep(0.5)
+
+        client = pymongo.MongoClient(results_cache.get_uri())
```

Comment:
```suggestion
        with pymongo.MongoClient(results_cache.get_uri()) as client:
```

I haven't tested this works, but it should.